pkgs/development/python-modules/withings-api/default.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, poetry-core
+, arrow
+, requests-oauthlib
+, typing-extensions
+, pydantic
+, responses
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "withings-api";
+  version = "2.4.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "vangorra";
+    repo = "python_withings_api";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-8cOLHYnodPGk1b1n6xbVyW2iju3cG6MgnzYTKDsP/nw=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'requests-oauth = ">=0.4.1"' '''
+    substituteInPlace pyproject.toml \
+      --replace 'addopts = "--capture no --cov ./withings_api --cov-report html:build/coverage_report --cov-report term --cov-report xml:build/coverage.xml"' '''
+  '';
+
+  propagatedBuildInputs = [
+    poetry-core
```

The build system is only used for building. Not need to propagate it.

```suggestion
  nativeBuildInputs = [
    poetry-core
  ];

  propagatedBuildInputs = [
```